server/src/main/java/org/apache/iotdb/db/metadata/MManager.java

```diff
@@ -313,23 +279,52 @@ private int applyMlog(MLogReader mLogReader) {
     return idx;
   }
 
+  private void checkMTreeModified() {
+    if (logWriter == null || logFile == null) {
+      // the logWriter is not initialized now, we skip the check once.
+      return;
+    }
+    if (System.currentTimeMillis() - logFile.lastModified() < mtreeSnapshotThresholdTime) {
+      if (logger.isDebugEnabled()) {
+        logger.debug(
+            "MTree snapshot need not be created. Time from last modification: {} ms.",
+            System.currentTimeMillis() - logFile.lastModified());
+      }
+    } else if (logWriter.getLogNum() < mtreeSnapshotInterval) {
+      if (logger.isDebugEnabled()) {
+        logger.debug(
+            "MTree snapshot need not be created. New mlog line number: {}.", logWriter.getLogNum());
+      }
+    } else {
+      logger.info(
+          "New mlog line number: {}, time from last modification: {} ms",
+          logWriter.getLogNum(),
+          System.currentTimeMillis() - logFile.lastModified());
+      createMTreeSnapshot();
+    }
+  }
+
+  public void createMTreeSnapshot() {
+    try {
+      mtree.createSnapshot();
+      logWriter.clear();
+    } catch (IOException e) {
+      logger.warn("Failed to create MTree snapshot", e);
+    }
+  }
+
   /** function for clearing MTree */
   public void clear() {
     try {
-      templateMap.clear();
-      this.mtree = new MTree();
+      this.mtree.clear();
```

Comment:
if this.mtree is null, will throw nullpointexception.

Reply:
Thanks for reminding.

Reply:
Fixed